docs/configuration.md

```diff
@@ -0,0 +1,541 @@
+---
+custom_edit_url: https://github.com/stryker-mutator/stryker-net/edit/master/docs/configuration.md
+---
+
+## Basics
+
+You run stryker from the test project directory.
+
+On some dotnet core projects stryker can run without specifying any custom configuration. Simply run `dotnet stryker` to start testing.  
+On dotnet framework projects the solution path argument is always required. Run at least `dotnet stryker --solution <solution-path>` or specify the solution file path in the config file to start testing. See [solution](#solution-<path>).
+
+## Use a config file
+When using Stryker regularly we recommend using a config file. This way you won't have to document how to run Stryker, you can save the config file in version control. To use a config file create a file called `stryker-config.json` in the (unit test) project folder and add a configuration section called stryker-config. Alternatively use [init](#init-<bool>)
+
+Example `stryker-config.json` file:
+``` javascript
+{
+    "stryker-config":
+    {
+        "solution": "../SolutionFile.sln",
+        "project": "ExampleProject.csproj"
+    }
+}
+```
+
+### `init` <`flag`>
+
+Default: `false`  
+Command line: `--init`  
+Config file: `N/A`  
+
+Creates a stryker config file with default options. Any options passed in the cli will override the defaults.
```

Comment:
Question about init. Will it override existing config files? Or throw an exception.

Reply:
init writes the config file after loading all config, so it will first load an existing config file, then patch with commandline options, then write back to disk and overwrite. I don't see a reason to throw an exception in this case, we will simply modify your config with new values and keep existing values in place. This change will then also show up in your git diff.

Reply:
The description for this option can be improved with this explaination.

Reply:
Exactly my thoughts ;)